pkg/controllers/managementuser/clusterauthtoken/common/user.go

```diff
@@ -31,17 +31,18 @@ func NewClusterAuthToken(token accessor.TokenAccessor, hashedValue string) *clus
 // NewClusterAuthSecret creates a new secret from the given token and its hash value
 // The cluster auth token is managed separately.
 // Does not create the secret in the remote cluster.
-func NewClusterAuthTokenSecret(token accessor.TokenAccessor, hashedValue string) *corev1.Secret {
-	return NewClusterAuthTokenSecretForName(token.GetName(), hashedValue)
+func NewClusterAuthTokenSecret(ns string, token accessor.TokenAccessor, hashedValue string) *corev1.Secret {
```

Comment:
Please check that the [`kube-api-auth`](https://github.com/rancher/kube-api-auth) proxy is not affected by these changes.
Or, if it is, create a PR adapting it to the changes.
Reason: It uses some of the common/ functions in its own functionality.

Reply:
Nice catch, this is indeed the only change that will require to be adapted after this PR is merged (aside from a bunch of dependency updates).
The rest of required changes are in [this PR](https://github.com/rancher/kube-api-auth/pull/38#), and can be merged without having to wait for this.